ipyleaflet/leaflet.py

```diff
@@ -32,35 +34,28 @@
                   'grabbing', 'help', 'no-drop', 'not-allowed', 'pointer',
                   'progress', 'text', 'wait', 'zoom-in', 'zoom-out']
 
+yesterday = (date.today() - timedelta(days=1)).strftime("%Y-%m-%d")
 
-def basemap_to_tiles(basemap, day='yesterday', **kwargs):
+
+def basemap_to_tiles(basemap, day=yesterday, **kwargs):
     """Turn a basemap into a TileLayer object.
 
     Parameters
     ----------
-    basemap : dict
+    basemap : class:`xyzservices.lib.TileProvider`
         Basemap description coming from ipyleaflet.basemaps.
     day: string
         If relevant for the chosen basemap, you can specify the day for
-        the tiles in the "%Y-%m-%d" format. Defaults to "yesterday".
+        the tiles in the "%Y-%m-%d" format. Defaults to yesterday's date.
     kwargs: key-word arguments
         Extra key-word arguments to pass to the TileLayer constructor.
     """
-    from datetime import date, timedelta
-
-    if day == 'yesterday':
-        yesterday = date.today() - timedelta(1)
-        day = yesterday.strftime('%Y-%m-%d')
-
-    url = basemap.get('url', '')
-    if url.count('%'):
-        url = url % day
-
```

Comment:
This will be a breaking change for people who were manually doing the string formatting, for example this Notebook does not work anymore: https://github.com/jupyter-widgets/tutorial/blob/master/notebooks/extras/ipyleaflet.ipynb.

(just a note that we shouldn't make a patch release with this)

Reply:
I think string formatting with different placeholders, can not be made backward compatible.